### PR TITLE
Feature/search files

### PR DESCRIPTION
**Pull request by vidalmelero**
New options added:

1. Project explorer search

2. Sort GO by name

3. "Select children" option

To sort GO by name, right click on any GO on the hierarchy, and select sort by name.

### CODE REVIEW

File: Engine/Module/ModuleScene.cpp [hunk 1]
```diff
@@ -334,6 +334,37 @@ Component * ModuleScene::GetComponent(uint64_t UUID) const
...
+		return name <= name2;
+	});
+
+	std::vector <GameObject*> game_objects;
+	for (auto go : list) 
+	{
+		game_objects.push_back(go);
+	}
+
+	go->children.swap(game_objects);
```
Comment (Dartemiss) [inline Engine/Module/ModuleScene.cpp:361]: Why do you swap?

**Review by Dartemiss — APPROVED** [review]
It works fine! But I thought this was another thing :(
I thought that the search files would allow searching actual files instead of directories, that would be very very useful, do you think it is very difficult to do it for the future?

GJ anyways

File: Engine/Module/ModuleScene.h [hunk 2]
```diff
@@ -39,7 +39,7 @@ class ModuleScene : public Module
 	ENGINE_API std::vector<GameObject*> GetGameObjectsWithTag(const std::string& tag) const;
 
 	Component* GetComponent(uint64_t UUID) const;
-
+	void SortGameObjectChilds(GameObject *go) const;
```
Comment (marcorod94) [inline Engine/Module/ModuleScene.h]: Code style

**Review by marcorod94 — APPROVED** [review]
Nice work hopefully you could search for files instead of directories

**Pull request merged**